options/core/src/main/java/polanski/option/OptionAssertion.java

```diff
@@ -0,0 +1,63 @@
+package polanski.option;
+
+import android.support.annotation.NonNull;
+
+import polanski.option.function.Func1;
+
+public final class OptionAssertion<T> {
+
+    @NonNull
+    private final Option<T> actual;
+
+    OptionAssertion(@NonNull final Option<T> actual) {
```

Comment:
So long as we can only create this via `Option`, then we can be sure that this is non-null.